dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java

```diff
@@ -211,4 +221,22 @@ public void close(String cause) {
     public void stop(String cause) {
         close(cause);
     }
+
+    /**
+     * kill all yarn tasks which are running
+     */
+    public void killAllRunningYarnTasks() {
```

Comment:
I don't think this problem can be solved. When this happens, the task has not been submitted to yarn, and the logic is the same as that of master fault tolerance

Reply:
I think it should kill the local process and kill the yarn task when the master is fault-tolerant

Reply:
Agree with @JinyLeeChina, a better way to solve #8110 might better to find out why the worker go into fake death.
And what is the `fake death`? I guess in `fake death` the close method will not be executed? If the WorkerServer can close, then it will easy to go to death.

Reply:
> I think it should kill the local process and kill the yarn task when the master is fault-tolerant
It's related to the order, the master add worker into dead server and failover worker, and then worker check itslef dead and close. During the worker heartbeat interval, the worker may run a new task, but it has not yet responded. Therefore, the worker should also kill the local process to prevent omission.

Reply:
I will add the reasons for this.